app/models/user_vote.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+class UserVote < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :poll
+
+  validates :user_id, uniqueness: { scope: :poll_id }
```

Comment:
I add indexes, but stackoverflow suggest use both methods, because error-handling with model-based uniqueness validation easier and one indexed query is not high price. I think they have a point, two defence layer better than one.